src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -158,40 +361,222 @@ export default {
 			this.opened = !this.opened
 			this.$emit('update:open', this.opened)
 		},
-		callPreventStop(action, event) {
-			if (action) {
-				event.preventDefault()
-				event.stopPropagation()
-				action()
-			}
-		},
-		cancelEdit(e) {
+		cancelEdit() {
 			// remove the editing class
-			if (Array.isArray(this.item.classes)) {
-				this.item.classes = this.item.classes.filter(item => item !== 'editing')
-			}
-			this.item.edit.reset(e)
+			this.editing = false
 		},
-		// This is used to decide which outter element type to use
-		// li or router-link
-		navElement(item) {
-			if (item.router) {
-				let exact = item.router.exact
-				if (typeof item.router.exact === 'undefined') {
-					exact = true
-				}
+		onClick(event) {
+			this.$emit('click', event)
+		},
+		handleEdit() {
+			this.editing = true
+		},
+		handleRename() {
+			this.$emit('update:title', this.newTitle)
+			this.newTitle = ''
+			this.editing = false
+		},
+		handleUndo() {
+			this.$emit('undo')
+		}
+	}
+}
+</script>
 
-				return {
-					is: 'router-link',
-					tag: 'li',
-					to: item.router,
-					exact
+<style lang="scss" scoped>
+@import '~Fonts/scss/iconfont-vue';
+
+.app-navigation-entry {
+	position: relative;
+	min-height: $clickable-area;
+	width: 100%;
+	box-sizing: border-box;
+	display: flex;
+	flex-wrap: wrap;
```

Comment:
Not needed. We don't want the item to be wrapped, hm?
```suggestion
```

Reply:
Yes we do, because the children element is wrapped under the main line.
```
| icon - text  - counter - menu |
|           childrens           |
```